parser/parser.go

```diff
@@ -162,6 +160,87 @@ func (p *parser) parseCommaList(end tokenKind, elementKind string) (*token, ast.
 	}
 }
 
+func (p *parser) parseArgument() (*ast.Identifier, ast.Node, error) {
+	var id *ast.Identifier
+	if p.peek().kind == tokenIdentifier && p.doublePeek().kind == tokenOperator && p.doublePeek().data == "=" {
+		ident := p.pop()
+		var tmpID = ast.Identifier(ident.data)
+		id = &tmpID
+		p.pop() // "=" token
+	}
+	expr, err := p.parse(maxPrecedence)
+	if err != nil {
+		return nil, nil, err
+	}
+	return id, expr, nil
+}
+
+// TODO(sbarzowski) - this returned bool is weird
+// TODO(sbarzowski) - name - it's also used for parameters
+func (p *parser) parseArguments(end tokenKind, elementKind string) (*token, *ast.Arguments, bool, error) {
+	args := &ast.Arguments{}
+	gotComma := false
+	namedArgumentAdded := false
+	first := true
+	for {
```

Comment:
You could break this loop out into two by first parsing only positional and then only named arguments. With a little cleverness you could get rid of the `first` variable as well. IMO it'll be a little more readable, since your cutting down from 3 booleans being toggled in a single loop down to 1.

Reply:
I tried it and I was able to simplify it a bit, but I wasn't able to do what you propose nicely. Perhaps I'm missing something, but here are the problems:
* gotComma is needed as it, b/c we return it and save it in the AST. So however we could eliminate `gotComma` and `first` in favor of something like `readyForNextArgument` we would still need to reconstruct `gotComma`. 
* We don't know ahead of time if the next argument is going to be positional or not, so we would need to pass the first named argument to the second loop somehow and we would need to handle both cases in both loops anyway.